utils/service_discovery/abstract_config_store.py

```diff
@@ -0,0 +1,173 @@
+# std
+import logging
+import simplejson as json
+from os import path
+
+# 3p
+from urllib3.exceptions import TimeoutError
+
+# project
+from utils.checkfiles import get_check_class, get_auto_conf, get_auto_conf_images
+from utils.singleton import Singleton
+
+log = logging.getLogger(__name__)
+
+CONFIG_FROM_AUTOCONF = 'auto-configuration'
+CONFIG_FROM_TEMPLATE = 'template'
+SD_TEMPLATE_DIR = '/datadog/check_configs'
+
+
+class KeyNotFound(Exception):
+    pass
+
+
+class AbstractConfigStore(object):
+    """Singleton for config stores"""
+    __metaclass__ = Singleton
+
+    previous_config_index = None
+
+    def __init__(self, agentConfig):
+        self.client = None
+        self.agentConfig = agentConfig
+        self.settings = self._extract_settings(agentConfig)
+        self.client = self.get_client()
+        self.sd_template_dir = agentConfig.get('sd_template_dir')
+        self.AUTO_CONF_IMAGES = get_auto_conf_images(agentConfig)
+
+    @classmethod
+    def _drop(cls):
+        """Drop the config store instance"""
+        del cls._instances[cls]
```

Comment:
You should double check that it does the right thing. Also, if it's just used for tests, you should add comments to say so.